nixos/modules/system/boot/kernel.nix

```diff
@@ -35,7 +35,7 @@ in
     };
 
     boot.kernelPackages = mkOption {
-      default = pkgs.linuxPackages;
+      default = if stdenv.isi686 then pkgs.linuxKernel.packages.linux_5_10 else pkgs.linuxPackages;
```

Comment:
Oh, I originally thought that we'd define `linuxPackages` itself with an exception based on `stdenv.hostPlatform`, but I'm not really sure which approach to pick.

Reply:
Don't mind how we do it. Not really sure how best to do what you're suggested though, all of the various `linuxPackages`, `linuxKernel`, etc packages and aliases is a bit of a mess.

Reply:
I trust that there's only one place where the default kernel branch is defined, so my idea would mean
```diff
--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -530,3 +530,3 @@ in {
   packageAliases = {
-    linux_default = packages.linux_5_10;
+    linux_default = if stdenv.hostPlatform.isi686 then packages.linux_5_10 else packages.linux_5_15;
     # Update this when adding the newest kernel major version!
```

Reply:
Done.